addon/-private/system/store.js

```diff
@@ -1905,38 +1895,55 @@ Store = Service.extend({
       return;
     }
 
-    this.typeMapFor(internalModel.type).idToRecord[id] = internalModel;
+    internalModel.schema.recordMap.idToRecord[id] = internalModel;
 
     internalModel.setId(id);
   },
 
   /**
-    Returns a map of IDs to client IDs for a given type.
+    Returns a map of IDs to client IDs for a given modelClass.
 
     @method typeMapFor
     @private
-    @param {DS.Model} typeClass
-    @return {Object} typeMap
+    @param {DS.Model} modelClass
+    @return {Object} recordMap
   */
-  typeMapFor(typeClass) {
+  typeMapFor(modelClass) {
```

Comment:
@pangratz yes, but I suspect we leaked this publicly and should add a deprecation.